src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/controllers/v1/person/SentencesController.kt

```diff
@@ -36,4 +39,18 @@ class SentencesController(
 
     return response.data.paginateWith(page, perPage)
   }
+
+  @GetMapping("{encodedHmppsId}/sentences/latest-key-dates-and-adjustments")
+  fun getPersonLatestSentenceKeyDatesAndAdjustments(
+    @PathVariable encodedHmppsId: String,
+  ): Map<String, LatestSentenceKeyDatesAndAdjustments?> {
+    val hmppsId = encodedHmppsId.decodeUrlCharacters()
+    val response = getLatestSentenceKeyDatesAndAdjustmentsForPersonService.execute(hmppsId)
+
+    if (response.hasErrorCausedBy(UpstreamApiError.Type.ENTITY_NOT_FOUND, causedBy = UpstreamApi.NOMIS)) {
+      throw EntityNotFoundException("Could not find person with id: $hmppsId")
+    }
+
+    return mapOf("data" to response.data)
```

Comment:
Explicit data key? Doesn't this come with the Response object, which can be paginated?

Reply:
Nahhhhh, this returns a single object because it's information about the latest sentence.

It's not a list of objects so no using our `PaginationResponse` class or `Response` class as that'll return all the errors as well. `Response` class is for returning in services and gateways.